dev/AppNotifications/ShellLocalization.cpp

```
@@ -0,0 +1,265 @@
+    }
+
+    return hIcon;
+}
+
+HRESULT Microsoft::Windows::AppNotifications::ShellLocalization::RetrieveAssetsFromProcess(_Out_ Microsoft::Windows::AppNotifications::ShellLocalization::AppNotificationAssets& assets) noexcept try
+{
+    wil::unique_hicon hIcon{ RetrieveIconFromProcess() };
+
+    std::wstring notificationAppId{ Microsoft::Windows::AppNotifications::Helpers::RetrieveNotificationAppId() };
+
+    // path: C:\Users\<currentUser>\AppData\Local\Microsoft\WindowsAppSDK\{AppGUID}.png
+    std::wstring iconFilePath{ RetrieveLocalFolderPath().c_str() + c_backSlash + notificationAppId + c_pngExtension };
+    WriteHIconToPngFile(hIcon, iconFilePath.c_str());
+
+    assets.displayName = Microsoft::Windows::AppNotifications::Helpers::GetDisplayNameBasedOnProcessName();
+    assets.iconFilePath = iconFilePath;
+
+    // TODO: Clean up the icon file in case UnregisterAll wasn't called.
+
+    return S_OK;
+}
+CATCH_RETURN()
+
+// Do nothing. This is just a placeholder while the UDK is ingested with the proper API.
+HRESULT Microsoft::Windows::AppNotifications::ShellLocalization::RetrieveAssetsFromShortcut(_Out_ Microsoft::Windows::AppNotifications::ShellLocalization::AppNotificationAssets& /*assets*/)
+{
+    // THROW_HR_IF_MSG(E_UNEXPECTED, IsIconFileExtensionSupported(iconFilePath));
+
+    return E_NOTIMPL;
+}
+
+HRESULT Microsoft::Windows::AppNotifications::ShellLocalization::RetrieveDefaultAssets(_Out_ Microsoft::Windows::AppNotifications::ShellLocalization::AppNotificationAssets& assets) noexcept try
+{
+    assets.displayName = Microsoft::Windows::AppNotifications::Helpers::GetDisplayNameBasedOnProcessName();
+    assets.iconFilePath = L""; // No icon filepath for now. This is coming in a future FrameworkUdk ingestion.
+
+    return S_OK;
+}
+CATCH_RETURN()
+
+void Microsoft::Windows::AppNotifications::ShellLocalization::DeleteIconFromCache() noexcept try
```

Don't swallow exceptions in methods that don't return HRESULT. Preferrably handle the exception at the caller by ignoring it, logging it or rethrow.